src/Nuget.PackageIndex.VisualStudio/Analyzers/ProjectTargetFrameworkProviderExporter.cs

```diff
@@ -175,6 +176,16 @@ private void OnAfterSolutionClosing()
             ClearCache();
         }
 
+        private void OnProjectRemoved(Project project)
+        {
+            if (project == null)
+            {
+                return;
+            }
+
+            OnProjectTargetFrameworkChanged(this, project.FullName);
+        }
+
```

Comment:
I haven't see a destructor in years (it took me a few minutes to remember the spelling :) )